3rdparty/openvx/include/openvx_hal.hpp

```diff
@@ -151,30 +173,171 @@ struct vxImage
+        e.print();
+        return CV_HAL_ERROR_UNKNOWN;
+    }
+    return CV_HAL_ERROR_OK;
+}
+
+#if defined OPENCV_IMGPROC_HAL_INTERFACE_H
+#define CV_HAL_INTER_NEAREST 0
+#define CV_HAL_INTER_LINEAR 1
+#define CV_HAL_INTER_CUBIC 2
+#define CV_HAL_INTER_AREA 3
+#define CV_HAL_INTER_LANCZOS4 4
+#define MORPH_ERODE 0
+#define MORPH_DILATE 1
+
+inline int ovx_hal_resize(int atype, const uchar *a, size_t astep, int aw, int ah, uchar *b, size_t bstep, int bw, int bh, double inv_scale_x, double inv_scale_y, int interpolation)
+{
+    try
+    {
+        vxContext * ctx = vxContext::getContext();
+        vxImage ia(*ctx, a, astep, aw, ah);
+        vxImage ib(*ctx, b, bstep, bw, bh);
+
+        if(!((atype == CV_8UC1 || atype == CV_8SC1) &&
+             inv_scale_x > 0 && inv_scale_y > 0 &&
+             (bw - 0.5) / inv_scale_x - 0.5 < aw && (bh - 0.5) / inv_scale_y - 0.5 < ah &&
+             (bw + 0.5) / inv_scale_x + 0.5 >= aw && (bh + 0.5) / inv_scale_y + 0.5 >= ah &&
+             std::abs(bw / inv_scale_x - aw) < 0.1 && std::abs(bh / inv_scale_y - ah) < 0.1 ))
+            vxErr(VX_ERROR_INVALID_PARAMETERS, "Bad scale").check();
+
+        int mode;
+        if (interpolation == CV_HAL_INTER_LINEAR)
+            mode = VX_INTERPOLATION_BILINEAR;
+        else if (interpolation == CV_HAL_INTER_AREA)
+            mode = VX_INTERPOLATION_AREA;
+        else if (interpolation == CV_HAL_INTER_NEAREST)
+            mode = VX_INTERPOLATION_NEAREST_NEIGHBOR;
+        else
+            vxErr(VX_ERROR_INVALID_PARAMETERS, "Bad interpolation mode").check();
+
+        vxErr::check( vxuScaleImage(ctx->ctx, ia.img, ib.img, mode));
+    }
+    catch (vxErr & e)
+    {
+        e.print();
+        return CV_HAL_ERROR_UNKNOWN;
+    }
+    return CV_HAL_ERROR_OK;
+}
+
+inline int ovx_hal_warpAffine(int atype, const uchar *a, size_t astep, int aw, int ah, uchar *b, size_t bstep, int bw, int bh, const double M[6], int interpolation, int, const double*)
+{
+    try
+    {
+        vxContext * ctx = vxContext::getContext();
+        vxImage ia(*ctx, a, astep, aw, ah);
+        vxImage ib(*ctx, b, bstep, bw, bh);
+
+        if (!(atype == CV_8UC1 || atype == CV_8SC1))
+            vxErr(VX_ERROR_INVALID_PARAMETERS, "Bad input type").check();
+
+        // It make sense to check border mode as well, but it's impossible to set border mode for immediate OpenVX calls
+        // So the only supported modes should be UNDEFINED(there is no such for HAL) and probably ISOLATED
+
+        int mode;
+        if (interpolation == CV_HAL_INTER_LINEAR)
+            mode = VX_INTERPOLATION_BILINEAR;
+        else if (interpolation == CV_HAL_INTER_AREA)
+            mode = VX_INTERPOLATION_AREA;
+        else if (interpolation == CV_HAL_INTER_NEAREST)
+            mode = VX_INTERPOLATION_NEAREST_NEIGHBOR;
+        else
+            vxErr(VX_ERROR_INVALID_PARAMETERS, "Bad interpolation mode").check();
+
+        vxMatrix mtx(*ctx, std::vector<float>(M, M + 6).data(), 2, 3);
```

Comment:
As far as I remember OpenVX and OpenCV uses different formulas for Warp operations (matrix should be transposed).